code/Control/UserDefinedFormController.php

```diff
@@ -182,11 +182,13 @@ public function generateConditionalJavascript()
         // Only add customScript if $default or $rules is defined
         if ($rules) {
             Requirements::customScript(<<<JS
-                (function($) {
+                function initUDFRules() {
+                    if (!window.jQuery) return setTimeout(initUDFRules, 50);
```

Comment:
I don't really like the "wait for 50ms" part of this change. Is there a better way we can fix this?

Reply:
I tried to find a different way but didn't find anything that would be simple yet working.
If jQuery emitted an event we could hook to via https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener, that could work too.

Reply:
What about a standard document ready callback?

Reply:
It's not a problem of waiting for the document to be ready, but waiting for jQuery to be available. This issue occurs when the rules are processed before jQuery is loaded (either placed after the rules or loaded using async).

Reply:
I can work around this issue by adding all the custom JS to the concatenation. Then it will always load after jQuery so it will work ok - if this is not the type of code you want to include.

Reply:
One could poll for `window.jQuery` with `setTimeout` - although very inelegant it would probably work.

The way in history to achieve this has been to enforce jQuery to load before this snippet via a preceding `Requirements::javascript` call to load it. This is probably no longer desirable in that it enforces a particular type of jQuery... but then this whole thing enforces jQuery anyway so...

Would we prefer to update it to AMD or something?

Reply:
>Would we prefer to update it to AMD or something?

That'd be a much bigger change than this is

Reply:
> One could poll for window.jQuery with setTimeout - although very inelegant it would probably work.

Nearly the same what the suggested change was doing - re-run the initiation waiting for jQuery to be available.

But I understand that without having an elegant solution to this it's better to work around it and make sure jQuery is available. Thanks for consideration anyway ;-)